dkpro-meta-example-galaxy/src/main/resources/templates/toolEngine.xml

```diff
@@ -50,19 +50,80 @@
       paramDecls.each { param ->
         def paramType = typeMap[param.type] ?: 'text';
         def paramValue = ( paramValues.find { it.name == param.name }?.value ) ?: ''; 
-    </gsp:scriptlet>
+	          	
+      	if (param.name.toLowerCase() == 'language') {
+        	paramType = 'select';
+ </gsp:scriptlet>
+        <conditional name="modelVariant">
+ <gsp:scriptlet>
+        } else if (param.name.toLowerCase().contains('modelvariant')) {
+       		return
+       	} else if (param.name.toLowerCase().contains('encoding')) {
+       		return
+       	} else if (param.name.toLowerCase().contains('location')) {
+       		return
+       	}  else if (param.name.toLowerCase().contains('overwrite')) {
+       		return
+       	}
+ </gsp:scriptlet>
         <param 
             name="${param.name}" 
             label="${param.name}" 
             type="${paramType}"
             optional="${!param.mandatory}"
             area="${param.multiValued}"
             value="${paramValue}"
-            help="${html2text(param.description)}">
-        </param>
-    <gsp:scriptlet>
+            help="${html2text(param.description)}"> 
+ <gsp:scriptlet>
+            if (param.name.toLowerCase() == 'language') {
+            	def langs = engine.languages as Set
+          		engine.models.each { model -> 
+          			def lang = (model.@language.text())
+          			langs.add(lang)
+          			}
+          		langs.each { lang ->
+ </gsp:scriptlet>
+			    	<option value="${lang}">$lang</option>
+ <gsp:scriptlet>
+			     }
+           	 } 
+ </gsp:scriptlet>
+		  	</param>
+ <gsp:scriptlet>
+		  	if (param.name == 'language') {
```

Comment:
There are three cases:
* a component has models - then this should be a dropdown based on the models
* a component has no models but declares language capabilities - then this dropdown should be based on the languages declared by the capabilities
* a component has a language parameter but not explicit information about which languages are supported - in this case, it should be a free text field.